### PR TITLE
feat(add-skills):add skill and find skill by user id api

### PR DESCRIPTION
**Pull request by karthikeyanpa12**
## Issue ID
None

## Summary
New Skills API

## Steps to implement this feature

Add New Table using PgAdmin.
```
CREATE TABLE public.t_skills(
    user_id varchar NOT NULL,
    skill varchar NOT NULL,
    skill_id int4 NOT NULL GENERATED ALWAYS AS IDENTITY
);
```

Add New DTO (Data Transfer Object) - SKillsRequest
```
// Adding lombok @Data
@Data
public class SkillsRequest {

}
```

Add New Model - Skills
```
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
@Table(name="t_skills")
public class Skills {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long skill_id;
}
```

Add New Service Interface - Skills.
```
public interface SkillsService {
  List<Profile> createSkills(SkillsRequest skillsRequest);
}
```

Add New JPA Repository interface.
```
public interface SkillsRepository extends JpaRepository<Skills, Long> {
  public List<Skills> findByUserId(String userId);
}
```

Add New Service Implementation - Skills.
```
// Autowire the JPA Repository.

@Service
public class SkillsServiceImpl implements SkillsService {

    @Autowired
    SkillsRepository skillsRepository;

    @Override
    public List<Skills> createSkills(SkillsRequest skillsRequest) {
        Skills skillsPayload = new Skills();
        skillsPayload.setUserId(skillsRequest.getUserId());
        skillsPayload.setSkill(skillsRequest.getSkill());
        skillsRepository.save(skillsPayload);
      return null;
    }
}
```

Add New Controller - skills.
```
// Update it as `@RestController`
// Add post api method using `@PostMapping(value = "/skills")` - should be double quoted value.
// Autowired the `SkillsService`.
// Creating method with `ResponseEntity<ApiResponse>` for adding new skill and throws `JsonParseException`.
// Validate user id is not empty.

@RestController
public class skills {

    @Autowired
    SkillsService skillsService;

    @PostMapping(value = "/skills")
    public ResponseEntity<ApiResponse> createSkills(@RequestBody SkillsRequest skillsRequest) throws JsonProcessingException {
        ApiResponse apiResponse = new ApiResponse();

        if (skillsRequest != null && StringUtils.isNotBlank(skillsRequest.getUserId())) {
            List<Skills> newSkill = skillsService.createSkills(skillsRequest);
            apiResponse.setResponse(newSkill);
            apiResponse.setStatus(HttpStatus.OK.toString());
            apiResponse.setMessage("new skills created");
        } else {
            apiResponse.setStatus(HttpStatus.BAD_REQUEST.toString());
            apiResponse.setMessage(Constants.EMPTY_USER_REQUEST);
        }
        return ResponseEntity.ok().body(apiResponse);
    }
}
```
Postman Screenshot 
Screenshot 2022-04-23 at 11 14 28 PM

  
Screenshot 2022-04-23 at 11 14 49 PM

## Checklist

- [X] Have you added the summary of what your changes do and why you'd like us to include them?
- [X] Have you added the steps to implement this feature?
- [ ] Have the files been linted and formatted?
- [ ] Have the docs been updated to match the changes in the PR?
- [ ] Have the tests been updated to match the changes in the PR?
- [ ] Have you run the tests locally to confirm they pass?

### What is the need for this tech update, and the steps to reproduce the issue?
User able to add new Skills.

### What is the current status?
No Option to add new skill to user

### How does this PR fix the problem, Screenshot Please?

### CODE REVIEW

**Comment by ramsunvtech** [comment]
@karthikeyanpa12 change the PR Title. its not only a Add Skill. you added find skill by user Id

File: src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    SkillsRepository skillsRepository;
+
+    @Override
+    public List<Skills> getSkillsByUserId(String userId) {
+        return skillsRepository.findByUserId(userId);
+    }
+
+    @Override
+    public List<Skills> createSkills(SkillsRequest skillsRequest) {
+        Skills skillsPayload = new Skills();
```
Comment (ramsunvtech) [inline src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java:25]: change this to `newSkill`

**Comment by ramsunvtech** [comment]
@karthikeyanpa12 get the current Postman Collection from https://github.com/ramsunvtech/linkedin-clone-api/blob/main/Postman%20Collection/LinkedInClone.postman_collection.json

Add your folder `experience`, `skills` and update the API

File: src/main/java/com/linkedin/ProfessionalNetworking/api/skills.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+public class skills {
```
Comment (jpukg) [inline src/main/java/com/linkedin/ProfessionalNetworking/api/skills.java:22]: Rename it to SkillController

File: src/main/java/com/linkedin/ProfessionalNetworking/model/Skills.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
```
Comment (jpukg) [inline src/main/java/com/linkedin/ProfessionalNetworking/model/Skills.java:11]: Remove extra annotations as these are not necessary in this place. `@Data` is sufficient.

File: src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        skillsPayload.setUserId(skillsRequest.getUserId());
+        skillsPayload.setSkill(skillsRequest.getSkill());
+        skillsRepository.save(skillsPayload);
+      return null;
```
Comment (jpukg) [inline src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java:29]: return the same object after persisting.

File: src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class SkillsServiceImpl implements SkillsService {
+
+    @Autowired
+    SkillsRepository skillsRepository;
```
Comment (jpukg) [inline src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java:16]: Specify access modifier, either one of these (private / public or protected)

File: src/main/java/com/linkedin/ProfessionalNetworking/api/skills.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Autowired
+    SkillsService skillsService;
```
Comment (jpukg) [inline src/main/java/com/linkedin/ProfessionalNetworking/api/skills.java:25]: Specify access modifier, either one of these (private / public or protected)

File: src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java (hunk 1 above)
Comment (karthikeyanpa12) [reply src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java:25]: done

File: src/main/java/com/linkedin/ProfessionalNetworking/api/skills.java (hunk 2 above)
Comment (karthikeyanpa12) [reply src/main/java/com/linkedin/ProfessionalNetworking/api/skills.java]: done

File: src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java (hunk 4 above)
Comment (karthikeyanpa12) [reply src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java]: done

File: src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java (hunk 5 above)
Comment (karthikeyanpa12) [reply src/main/java/com/linkedin/ProfessionalNetworking/service/impl/SkillsServiceImpl.java]: done

File: src/main/java/com/linkedin/ProfessionalNetworking/api/skills.java (hunk 6 above)
Comment (karthikeyanpa12) [reply src/main/java/com/linkedin/ProfessionalNetworking/api/skills.java]: done

**Review by ramsunvtech — APPROVED** [review]
LGTM

**Pull request merged**